grouper/service_account.py

```diff
@@ -24,30 +24,54 @@
 from grouper.models.service_account import ServiceAccount
 from grouper.models.service_account_permission_map import ServiceAccountPermissionMap
 from grouper.models.user import User
+from grouper.plugin import get_plugins, PluginRejectedMachineSet
 from grouper.user import disable_user, enable_user
 
 # A single service account permission.
 ServiceAccountPermission = namedtuple("ServiceAccountPermission",
     ["permission", "argument", "granted_on", "mapping_id"])
 
 
+class BadMachineSet(Exception):
+    """The service account machine set was rejected."""
+    pass
+
+
 class DuplicateServiceAccount(Exception):
     """Creating a service account failed because it duplicates an existing user."""
     pass
 
 
+def _check_machine_set(service_account, machine_set):
+    # type: (ServiceAccount, str) -> None
+    """Verify a service account machine set with plugins.
+
+    Raises:
+        BadMachineSet: if some plugin rejected the machine set
+    """
+    try:
+        for plugin in get_plugins():
+            plugin.check_machine_set(service_account.user.username, machine_set)
+    except PluginRejectedMachineSet as e:
+        raise BadMachineSet(str(e))
```

Comment:
for `PluginRejectedGroupMembershipUpdate` and `PluginRejectedDisablingUser` we bubbled the exception all the way up. this is a little more overhead but I like the separation. perhaps we should do the same for the other plugin exceptions.

Reply:
Yeah, >>TODO.